include/ccl_core.h

```diff
@@ -11,6 +11,37 @@
 
 CCL_BEGIN_DECLS
 
+
+//p2d extrapolation types for early times 
+typedef enum ccl_p2d_extrap_growth_t
+{
+  ccl_p2d_cclgrowth = 401, //Use CCL's linear growth
+  ccl_p2d_customgrowth = 402, //Use a custom growth function
+  ccl_p2d_constantgrowth = 403, //Use a constant growth factor
+  ccl_p2d_no_extrapol = 404, //Do not extrapolate, just throw an exception
+} ccl_p2d_extrap_growth_t;
+
+//p2d interpolation types
+typedef enum ccl_p2d_interp_t
+{
+  ccl_p2d_3 = 303, //Bicubic interpolation
+} ccl_p2d_interp_t;
+
+/**
+ * Struct containing a 2D power spectrum
+ */
+typedef struct {
+  double lkmin,lkmax; /**< Edges in log(k)*/
+  double amin,amax; /**< Edges in a*/
+  int extrap_order_lok; /**< Order of extrapolating polynomial in log(k) for low k (0, 1 or 2)*/
+  int extrap_order_hik; /**< Order of extrapolating polynomial in log(k) for high k (0, 1 or 2)*/
+  ccl_p2d_extrap_growth_t extrap_linear_growth;  /**< Extrapolation type at high redshifts*/
+  int is_log; /**< Do I hold the values of log(P(k,a))?*/
+  double (*growth)(double); /**< Custom extrapolating growth function*/
+  double growth_factor_0; /**< Constant extrapolating growth factor*/
+  gsl_spline2d *pk; /**< Spline holding the values of P(k,a)*/
+} ccl_p2d_t;
+
```

Comment:
Whoops. I think we might have gotten confused between various comments. These should be in `ccl_ps2d.h` right? I only meant the ones you moved below.

Reply:
so a `ccl_cosmology` contains `ccl_p2d_t` structs, and at the same time the functions that use `ccl_p2d_t` structs also use ccl_cosmology objects. If I put the definition of `ccl_p2d_t` in `ccl_p2d.h` I get a circular dependence between `ccl_core.h` and `ccl_p2d.h`, which is very annoying

Reply:
ah shit. well maybe dump the extra header file or just leave it.